datafusion/src/physical_plan/expressions/average.rs

```diff
@@ -37,8 +37,8 @@ use super::{format_state_name, sum};
 #[derive(Debug)]
 pub struct Avg {
     name: String,
+    #[allow(dead_code)]
```

Comment:
I will look into removing this field as well

Reply:
This PR might be the reason causes the different behavior on `dead_code`. 
https://github.com/rust-lang/rust/pull/85200

Reply:
I think you are right